extensions/objects/templates/image-with-regions-editor.component.ts

```diff
@@ -314,6 +314,16 @@ export class ImageWithRegionsEditorComponent implements OnInit {
   }
 
   getPreviewUrl(imageUrl: string): string {
+    if (
+      this.contextService.getEntityType() !== (
+        AppConstants.ENTITY_TYPE.EXPLORATION)
+    ) {
+      const entityType: string = this.contextService.getEntityType() || '';
```

Comment:
I'm worried about this. '' does not seem like a reasonable value for entityType...

Reply:
'' was added in order to prevent a lint error since the return type of `getEntityType()` is `string | undefined` but `assetsBackendApiService.getImageUrlForPreview()` only expects a string as `getEntityType()`.

Reply:
But that's still not OK. In what cases can undefined happen? We should try to prevent those cases.

Reply:
If the current path name doesn't include any of `create` `explore` `embed` `topic_editor` `story_editor` `skill_editor` `blog-dashboard` or no custom context entity is set, `getEntityType()` will return undefined. SInce a custome context entity is being set in this case, it should not return an undefine.

Reply:
Then I think you need to do something like -- getEntityType() (no backup of ''), and if it's undefined, throw an error.